sources/targets/Stride.Core.targets

```diff
@@ -40,20 +40,6 @@
     <AdditionalFiles Include="$(MSBuildThisFileDirectory)stylecop.json" />
     <PackageReference Include="StyleCop.Analyzers" PrivateAssets="All" />
   </ItemGroup>
-  <!-- Disable analyzers unless StrideEnableCodeAnalysis is set (note: PackageReference can't depend on external Condition) -->
-  <Target Name="DisableAnalyzersForStrideBuild"
-        BeforeTargets="CoreCompile"
-        Condition="'$(StrideEnableCodeAnalysis)' != 'true'">
-    <ItemGroup>
-      <!-- We want to include Stride analyzers by default -->
-      <AnalyzersNotToRemove Include="@(Analyzer)" Condition="$([System.String]::Copy(%(FullPath)).Contains('Stride'))" />
-      <!-- We want to include source generators by default -->
-      <AnalyzersNotToRemove Include="@(Analyzer)" Condition="$([System.String]::Copy(%(FullPath)).Contains('Generator'))" />
-      <AnalyzersNotToRemove Include="@(Analyzer)" Condition="$([System.String]::Copy(%(FullPath)).Contains('SourceGeneration'))" />
-      <AnalyzersToRemove Include="@(Analyzer)" Exclude="@(AnalyzersNotToRemove)" />
-      <Analyzer Remove="@(AnalyzersToRemove)"/>
-    </ItemGroup>
-  </Target>
```

Comment:
This shouldn't be removed.

Reply:
it doesnt make a difference, stride compiles fine without it


i dont know why it is there

Reply:
Don't make changes that are not needed. It was there for a reason, and unless your changes require it to be removed it shouldn't. A PR should have the minimal changes that are required for a given feature, nothing more.

If there is an argument to be made about removing that part, it should be on its own PR and discussed there.
If as you said "it doesn't make a difference", then leave it.

I believe one of the reason this was added was to speed-up the build by disabling analysis that is not required  (esp. during CI), as well as removing a lot of warnings that would make the logs harder to read in case a certain build needs investigating.

Reply:
After discussion on Discord, it appears that the Stylecop settings might be some Xenko leftovers. Maybe @xen2 can shine a light on this. But basically:
1. we are adding a package reference for Stylecop assembly for all projects that have `<StrideCodeAnalysis>true</StrideCodeAnalysis>`. This seems to be most core and engine projects (not applied to editor).
2. since these generates a lot of warnings in the build log, those analyzers are disabled by default (unless `StrideEnableCodeAnalysis` is also set to `true`.
3. however, some analyzers/generators still need to be enabled at all time. Hence this "complex" rule with a allow-list of analyzers to not remove.

We have two options:
1. Either we keep the stylecop rule, in which case we can add another exception for PolySharp in that "alow-list".
2. We decide we don't want stylecop anymore and we just remove all of that.
  a. we do keep the first `StrideCodeAnalysis` section to use the proper ruleset

Reply:
currently stylecop is added to each project, but no project has the StrideEnableCodeAnalysis so it gets removed immediately after

Reply:
I think it was initially meant as an optional command line parameters when running a build.
Something like: `msbuild -p:StrideEnableCodeAnalysis=true ...`

Anyway, if #2105 is approved and merged, we won't need to discuss it anymore :sweat_smile: